pkg/infrastructure/kogito_infra.go

```diff
@@ -63,3 +64,21 @@ func IsKafkaResource(instance *v1beta1.KogitoInfra) bool {
 func IsKnativeEventingResource(instance *v1beta1.KogitoInfra) bool {
 	return instance.Spec.Resource.APIVersion == KnativeEventingAPIVersion && instance.Spec.Resource.Kind == KnativeEventingBrokerKind
 }
+
+// RemoveKogitoInfraOwnership remove provided kogito service owner reference from kogitoInfra
+func RemoveKogitoInfraOwnership(client *client.Client, owner v1beta1.KogitoService) error {
+	log.Debugf("Removing KogitoInfra ownership for %s", owner.GetName())
+	for _, kogitoInfraName := range owner.GetSpec().GetInfra() {
+		// load infra resource
+		kogitoInfraInstance, err := MustFetchKogitoInfraInstance(client, kogitoInfraName, owner.GetNamespace())
+		if err != nil {
+			return err
```

Comment:
Do we need to throw an error in case KogitoInfra is not found? I think that in such case we can just skip it.
Otherwise when for example user deletes KogitoInfra first and then KogitoRuntime, the KogitoRuntime won't be removed as operator will get stuck in reconciliation cycle returning error.

Reply:
Agreed.

Reply:
@sutaakar Thanks for pointed it out.  I had incorporated your suggestion